[INFO   ] Logger: Record log in /storage/emulated/0/Documents/project_utopia/.kivy/logs/kivy_21-09-24_14.txt
[INFO   ] Kivy: v1.11.1
[INFO   ] Kivy: Installed at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/__init__.py"
[INFO   ] Python: v3.8.3 (default, May 27 2020, 02:08:17) 
[GCC 9.3.0]
[INFO   ] Python: Interpreter at "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/bin/python3"
[INFO   ] Factory: 184 symbols loaded
[INFO   ] Image: Providers: img_tex, img_dds, img_sdl2, img_gif (img_pil, img_ffpyplayer ignored)
[INFO   ] Text: Provider: sdl2
[INFO   ] Window: Provider: sdl2
[INFO   ] GL: Using the "OpenGL ES 2" graphics system
[INFO   ] GL: Backend used <sdl2>
[INFO   ] GL: OpenGL version <b'OpenGL ES 3.2 v1.r26p0-01eac0.###other-sha0123456789ABCDEF0###'>
[INFO   ] GL: OpenGL vendor <b'ARM'>
[INFO   ] GL: OpenGL renderer <b'Mali-G72'>
[INFO   ] GL: OpenGL parsed version: 3, 2
[INFO   ] GL: Texture max size <8192>
[INFO   ] GL: Texture max units <16>
[INFO   ] Window: auto add sdl2 input provider
[INFO   ] Window: virtual keyboard not allowed, single mode, not docked
[WARNING] Factory: Ignored class "TopBar" re-declaration. Current -  module: None, cls: <class '__main__.TopBar'>, baseclass: None, filename: None. Ignored -  module: None, cls: None, baseclass: BoxLayout, filename: /storage/emulated/0/Documents/project_utopia/utopia.kv.
[WARNING] Factory: Ignored class "GameArea" re-declaration. Current -  module: None, cls: <class '__main__.GameArea'>, baseclass: None, filename: None. Ignored -  module: None, cls: None, baseclass: GridLayout, filename: /storage/emulated/0/Documents/project_utopia/utopia.kv.
[WARNING] Factory: Ignored class "UtopiaGame" re-declaration. Current -  module: None, cls: <class '__main__.UtopiaGame'>, baseclass: None, filename: None. Ignored -  module: None, cls: None, baseclass: BoxLayout, filename: /storage/emulated/0/Documents/project_utopia/utopia.kv.
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 249, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10, in <module>
[WARNING] stderr:     food_count: Food
[WARNING] stderr: NameError: name 'Food' is not defined
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 690, in _apply_rule
[WARNING] stderr:     value, bound = create_handler(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 252, in create_handler
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: kivy.lang.builder.BuilderException: Parser: File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10:
[WARNING] stderr: ...
[WARNING] stderr:       8:    size_hint: 1, .1
[WARNING] stderr:       9:    orientation: 'horizontal'
[WARNING] stderr: >>   10:    food_count: Food
[WARNING] stderr:      11:    water_count: Water
[WARNING] stderr:      12:    magic_count: Magic
[WARNING] stderr: ...
[WARNING] stderr: NameError: name 'Food' is not defined
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 249, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10, in <module>
[WARNING] stderr:     food_count: Food
[WARNING] stderr: 
[WARNING] stderr: 
[WARNING] stderr: During handling of the above exception, another exception occurred:
[WARNING] stderr: 
[WARNING] stderr: Traceback (most recent call last):
[WARNING] stderr:   File "/storage/emulated/0/Documents/project_utopia/utopia_app.py", line 78, in <module>
[WARNING] stderr:     UtopiaApp().run()
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/app.py", line 828, in run
[WARNING] stderr:     self.load_kv(filename=self.kv_file)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/app.py", line 599, in load_kv
[WARNING] stderr:     root = Builder.load_file(rfilename)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 301, in load_file
[WARNING] stderr:     return self.load_string(data, **kwargs)
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 401, in load_string
[WARNING] stderr:     widget.apply_class_lang_rules(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/uix/widget.py", line 467, in apply_class_lang_rules
[WARNING] stderr:     Builder.apply(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 536, in apply
[WARNING] stderr:     self._apply_rule(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 656, in _apply_rule
[WARNING] stderr:     child.apply_class_lang_rules(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/uix/widget.py", line 467, in apply_class_lang_rules
[WARNING] stderr:     Builder.apply(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 536, in apply
[WARNING] stderr:     self._apply_rule(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 705, in _apply_rule
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: kivy.lang.builder.BuilderException: Parser: File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10:
[WARNING] stderr: ...
[WARNING] stderr:       8:    size_hint: 1, .1
[WARNING] stderr:       9:    orientation: 'horizontal'
[WARNING] stderr: >>   10:    food_count: Food
[WARNING] stderr:      11:    water_count: Water
[WARNING] stderr:      12:    magic_count: Magic
[WARNING] stderr: ...
[WARNING] stderr: BuilderException: Parser: File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10:
[WARNING] stderr: ...
[WARNING] stderr:       8:    size_hint: 1, .1
[WARNING] stderr:       9:    orientation: 'horizontal'
[WARNING] stderr: >>   10:    food_count: Food
[WARNING] stderr:      11:    water_count: Water
[WARNING] stderr:      12:    magic_count: Magic
[WARNING] stderr: ...
[WARNING] stderr: NameError: name 'Food' is not defined
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 249, in create_handler
[WARNING] stderr:     return eval(value, idmap), bound_list
[WARNING] stderr:   File "/storage/emulated/0/Documents/project_utopia/utopia.kv", line 10, in <module>
[WARNING] stderr:     food_count: Food
[WARNING] stderr: 
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 690, in _apply_rule
[WARNING] stderr:     value, bound = create_handler(
[WARNING] stderr:   File "/data/user/0/ru.iiec.pydroid3/files/aarch64-linux-android/lib/python3.8/site-packages/kivy/lang/builder.py", line 252, in create_handler
[WARNING] stderr:     raise BuilderException(rule.ctx, rule.line,
[WARNING] stderr: 
